node/src/service.rs

```diff
@@ -141,6 +173,40 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
 			&config, backend.clone(), task_manager.spawn_handle(), client.clone(), network.clone(),
 		);
 	}
+	
+	// Current best block at initialization, to report to the RPC layer.
+	let last_hash = client.info().finalized_hash;
+	let block_collected = format!("{}", last_hash.clone());
+	info!("Current block saved to peer DHT {:?}", block_collected);
+
+	//define batch key as_bytes
+	let current_block_number: u32 = client.info().finalized_number;
+	
+	//todo! make batch increment after every 100 count
+	let mut batch_number = 0;
+	if current_block_number == 100 {
+		batch_number += 1;
+	}
+
+	let parsed_block_number = format!("{}", batch_number.clone());
+
+	info!("DHT record key {:?}", &parsed_block_number);
+	let key = Key::from(parsed_block_number.as_bytes().to_vec());
+
+	let data = DataMap {
+		key: key,
+		value: block_collected.as_bytes().to_vec()
+	};
+
+	// parse data into dht 
+	network.put_value(data.key.clone(), data.value);
+
+	// get value from dht
+	network.get_value(&data.key.clone());
```

Comment:
- The same thing that we have done here need to implement in way so that we can interact with it via RPC call or in runtime.
- Also I think it will only run once while we run the node. So don't know what you will get as `finalized_number` or `finalized_hash`

Reply:
the finalized number is just the serial number of the block finalised(i.e just hashed), while the finalised hash is the hash, so it's on working with the runtime.